### PR TITLE
Update ArgoCD Integration to match the new capabilities on ArgoCD 2.3

### LINKED ISSUE #185: Age encryption in ArgoCD is not working: the URL scheme 'secrets+age-import' is not allowed
### Current Behavior

When I'm trying to add an application via ArgoCD UI or via CLI I'm getting error that "File does not exist"

Error log:
`[helm-secrets] File does not exist: secrets+age-import:///helm-secrets-private-keys/key.txt?secrets.yaml\\nError: plugin \\\"secrets\\\" exited with error\",\"type\":\"ComparisonError\"}]}}" application=wordpress`

```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: wordpress
spec:
  destination:
    name: ''
    namespace: wordpress
    server: 'https://kubernetes.default.svc'
  source:
    path: xyz/example-wordpress
    repoURL: '<REPO_URL>'
    targetRevision: init
    helm:
      valueFiles:
        - values.yaml
        - secrets+age-import:///helm-secrets-private-keys/key.txt?secrets.yaml
  project: default
```
project tree
```
example-wordpress$ tree .
.
├── Chart.yaml
├── secrets.yaml
└── values.yaml
```

I double-checked:
- volume with the secret is properly mounted and accessible
- executed `helm template . -f secrets+age-import:///helm-secrets-private-keys/key.txty?secrets.yaml` locally in repo-server pod and it works as expected, decrypted values are correct
- tried the approach with k8s secret for getting age key

### Expected Behavior

When I add the application in ArgoCD with values that are encrypted using age, helm secrets should decrypt it and install app with decrypted values

### Steps To Reproduce

_No response_

### Environment

- Helm Version: 3.6.0
- Helm Secrets Version: 3.12.0
- OS: Ubuntu 21.04
- Shell: bash, version 5.1.4(1)-release

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jkroepke**
**What this PR does / why we need it**: Update ArgoCD Integration to match the new capabilities on ArgoCD 2.3

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #185

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: docs/ArgoCD Integration.md [hunk 1]
```diff
@@ -75,8 +66,12 @@ ARG SOPS_VERSION="3.7.1"
 ARG HELM_SECRETS_VERSION="3.12.0"
 ARG KUBECTL_VERSION="1.22.0"
 
-ENV HELM_SECRETS_HELM_PATH=/usr/local/bin/helm
-ENV HELM_PLUGINS="/home/argocd/.local/share/helm/plugins/"
+# In case, wrapper scripts are used, HELM_SECRETS_HELM_PATH needs to be the path of the real helm binary
```
Comment (Enrico204) [inline docs/ArgoCD Integration.md:69]: Only a typo: the comma after "`In case`" is not needed: `In case wrapper scripts are used` is ok :-)

File: docs/ArgoCD Integration.md [hunk 2]
```diff
@@ -147,15 +157,17 @@ repoServer:
...
+When using private key encryption it is required to configure ArgoCD repo server so that it has access 
+to the private key to decrypt the encrypted value file(s). When using GCP KMS, encrypted value file(s)
+can be decrypted using [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
 
 ## Private key encryption
 There are two ways to configure ArgoCD to have access to your private key.
-Either mount the gpg key from secret as volume or access the fetch the private key directly from a kubernetes secret.
-Both methods depend on a secret holding the key in asci format.
+Either mount on the argocd-repo-server the gpg/age key from secret as volume or access the fetch the
+private key directly from a kubernetes secret. Both methods depend on a secret holding the key in ASCII format.
```
Comment (Enrico204) [inline docs/ArgoCD Integration.md:167]: I would rephrase this sentence like:

> There are two ways to configure ArgoCD to have access to your private key:
> 
> - mount the PGP/age key secret as a volume in the `argocd-repo-server`; or
> - fetch the secret value (private key) directly using Kubernetes API.
> 
> Both methods depend on a Kubernetes secret holding the key in plain-text format (i.e., not encrypted).

I assumed that for "ASCII" you mean "not encrypted". Correct me if I'm wrong :-)

Comment (jkroepke) [reply docs/ArgoCD Integration.md:167]: I mean the ASCII armored format here - https://unix.stackexchange.com/questions/623375/what-is-the-armored-option-for-in-gnupg.

Comment (Enrico204) [reply docs/ArgoCD Integration.md:167]: Ah ok, my bad :-D

**Comment by Enrico204** [comment]
LGTM!

**Review by Enrico204 — APPROVED** [review]
